js/app/src/Blocks.svelte

```diff
@@ -254,25 +255,38 @@
 		} else {
 			if (dep.backend_fn) {
 				if (dep.trigger_mode === "once") {
-					if (!dep.pending_request) make_prediction(payload);
+					if (!dep.pending_request)
+						make_prediction(payload, dep.connection == "stream");
 				} else if (dep.trigger_mode === "multiple") {
-					make_prediction(payload);
+					make_prediction(payload, dep.connection == "stream");
 				} else if (dep.trigger_mode === "always_last") {
 					if (!dep.pending_request) {
-						make_prediction(payload);
+						make_prediction(payload, dep.connection == "stream");
 					} else {
 						dep.final_event = payload;
 					}
 				}
 			}
 		}
 
-		async function make_prediction(payload: Payload): Promise<void> {
+		async function make_prediction(
```

Comment:
If we're streaming, `make_prediction` will send the latest stream event data to the queue as opposed to creating a whole new event.